Framework/Core/src/ExternalFairMQDeviceProxy.cxx

```diff
@@ -141,18 +141,17 @@ void sendOnChannel(FairMQDevice& device, FairMQMessagePtr&& headerMessage, FairM
   sendOnChannel(device, out, spec, channelRetriever);
 }
 
-InjectorFunction o2DataModelAdaptor(OutputSpec const& spec, uint64_t startTime, uint64_t step)
+InjectorFunction o2DataModelAdaptor(OutputSpec const& spec, uint64_t step)
 {
-  auto timesliceId = std::make_shared<size_t>(startTime);
-  return [timesliceId, step, spec](FairMQDevice& device, FairMQParts& parts, ChannelRetriever channelRetriever) {
+  return [step, spec](FairMQDevice& device, FairMQParts& parts, ChannelRetriever channelRetriever) {
     for (size_t i = 0; i < parts.Size() / 2; ++i) {
       auto dh = o2::header::get<DataHeader*>(parts.At(i * 2)->GetData());
 
-      DataProcessingHeader dph{*timesliceId, 0};
+      // Now that the first orbit is part of the DataHeader, we can use that if
+      // the DataHeader is present.
+      DataProcessingHeader dph{dh->firstTForbit, 0};
```

Comment:
So I don't prepare the processing header? That would make things simpler for me. 
Will you also strip the DataProcessingHeader on the proxy output?

Reply:
DataProcessingHeader had been introduced to implement the data flow in DPL. Maybe it can be dropped/ignored outside DPL.

We do not need to strip it in the output proxy. But the proxy should check that the DataHeader is consistent. So you can simply ignore DataProcessingHeader

Reply:
+1 for the check in the proxy. Actually ignoring stuff on the stack is not that easy. If it is indeed just a DPL flow helper, I would remove it as soon as it's not needed anymore. (not to mention that would decouple outside users from the o2::framework lib)
@mkrzewic any thoughts on this?

Reply:
`DataProcessingHeader` was always meant to be an internal detail until we could get a `DataHeader` which supported DPL. So, no objection on my side, especially if we get duration and finer grained orbit as well, so that the overhead / complexity of navigating the headers is not simply moved somewhere else.